x/stakeibc/keeper/community_pool_test.go

```diff
@@ -438,3 +439,95 @@ func (s *KeeperTestSuite) TestRedeemCommunityPoolTokens_Failure_NotEnoughDelegat
 	err := s.App.StakeibcKeeper.RedeemCommunityPoolTokens(s.Ctx, invalidHostZone)
 	s.Require().ErrorContains(err, "invalid amount")
 }
+
+// ----------------------------------------------------------
+//               BuildFundCommunityPoolMsg
+// ----------------------------------------------------------
+
+func (s *KeeperTestSuite) TestBuildFundCommunityPoolMsg() {
+	withdrawalICA := "withdrawal_ica"
+	communityPoolReturnICA := "community_pool_return_ica"
+	communityPoolTreasuryAddress := "community_pool_treasury"
+
+	testCases := []struct {
+		name             string
+		senderAccoutType types.ICAAccountType
+		sendToTreasury   bool
+		expectedSender   string
+		expectedReceiver string
+		expectedError    string
+	}{
+		{
+			name:             "community pool return ICA to main community pool",
```

Comment:
are withdrawalica and community pool return ica the only possible senders here?

Reply:
yup, at least with the current implementation (easy to extend it later if needed, but need custom code for each address)